### PR TITLE
Create evms_testnets and evms_mainnets_testnets

### PR DESCRIPTION
**Pull request by MSilb7**
This spell creates `evms_testnets` (a fork of evms) to hold testnet chains (L1 Goerli + eventual future testnets) + a unified `evms_mainnets_testnets` that holds both mainnet and testnet chains.

~**(Forked from @hildobby's dunesql migration here #3640 -which should be merged before this PR )**~ DONE

NOTE: **My preferred solution is to just add testnets to `evms`** and add a column (or lookup table) indicating if a chain is a mainnet, testnet or something else. But porting it out, per discussions here #3648 and concerns about confusion among wizards who may expect mainnets only. A con is that this solution adds a ton of tech debt for any future changes or iterations, and likely results in testnets being left out of spells that are built on top of `evms`.

This is a proof of concept (it should work), but my personal vote is to merge testnets in to the `evms` schema per here: #3607 and avoid sprawling tables.

# SPELLBOOK FREEZE

From June 22 to June 29, we will be freezing some Spellbook contributions for the migration to DuneSQL. We will not accept contributions to the lineage of the following models:

* dex.trades
* nft.trades
* labels
* token.erc20
* tokens.nft

Run the following command to see the list of affected files:

```
dbt ls --resource-type model --output path --select +dex_trades +labels +nft_trades +tokens_nft +tokens_erc20
```

Don't hesitate to reach out on Discord if you have any questions.

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Review by alokeutpal — APPROVED** [review]

**Comment by MSilb7** [comment]
Tests passed, but cc @0xRobin you may have macro ideas to make this better. My thought is "can we store chain names in one spot, rather than duplicating them across all table spells"

Note that there is L2 logic to be cleaned up in this PR: https://github.com/duneanalytics/spellbook/pull/3608

**Comment by MSilb7** [comment]
^ aside from that comment, this is good @jeff-dude

Main purpose is add Goerli (and any future testnets) to evms in a way that doesn't confuse people who may use evms to mean "mainnets" (per @hildobby's recommendation).

I use mainnets + testnets together for developer metrics (see: https://dune.com/queries/2382060/3906387)

File: models/evms/evms_transactions.sql [hunk 1]
```diff
@@ -50,7 +50,7 @@ FROM (
         , CAST(NULL AS DECIMAL(38,0)) AS l1_fee
         , CAST(NULL AS DECIMAL(38,0)) AS l1_gas_price
         , CAST(NULL AS DECIMAL(38,0)) AS l1_gas_used
-        , NULL AS l1_timestamp
+        , cast(NULL as bigint) AS l1_timestamp
```
Comment (jeff-dude) [inline models/evms/evms_transactions.sql:53]: found it interesting to see this as `bigint`, but confirmed in data explorer

**Comment by jeff-dude** [comment]
it probably is worth it to create a macro to use universally for mainnet/testnet. could even accept columns as params potentially and have just one macro. if you want help with that from our side, it may take some time as we need to focus on key migration items and clearing out the PR queue

File: models/evms_mainnets_testnets/evms_mainnets_testnets_blocks.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     ('mainnet', ref('evms_blocks'))
+     ,('testnet', ref('evms_testnets_blocks'))
+] %}
+
+SELECT *
+FROM (
+        {% for model in models %}
+        SELECT
+        '{{ model[0] }}' AS chain_type
+        , *
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/evms_mainnets_testnets/evms_mainnets_testnets_blocks.sql:22]: Yeah, the assumption is that mainnets and testnets should have the same columns.

The "real" solution may be some macro that contains the columns and these all just reference that?

**Comment by MSilb7** [comment]
> it probably is worth it to create a macro to use universally for mainnet/testnet. could even accept columns as params potentially and have just one macro. if you want help with that from our side, it may take some time as we need to focus on key migration items and clearing out the PR queue

ha yeah had the same idea.

I didn't want to blow this out, but I could modify this PR to do the macros. This isn't super urgent.

**Comment by jeff-dude** [comment]
> > it probably is worth it to create a macro to use universally for mainnet/testnet. could even accept columns as params potentially and have just one macro. if you want help with that from our side, it may take some time as we need to focus on key migration items and clearing out the PR queue
> 
> ha yeah had the same idea.
> 
> I didn't want to blow this out, but I could modify this PR to do the macros. This isn't super urgent.

since it isn't urgent, plz do go ahead and try that. there should be some good examples out there in the macros directory.

**Comment by jeff-dude** [comment]
closing this for now, we can revisit after more active PRs are resolved

**Pull request closed without merging**